app/src/pages/SectionEdit.tsx

```diff
@@ -35,130 +54,155 @@ const SectionEdit = () => {
 
     // Fetch section details
     const { data: sectionData, error: sectionError } = useSWR(
-        token ? [`${BACKEND_URL}/sections/${sid}`, token] : null,
+        token ? [`${BACKEND_URL}/api/sections/${sid}`, token] : null,
         SectionServices.getSectionDetail
     );
 
+
+    // Fetch the exercises data from the server.
+    /*
+    const { data: exerciseData, error: exerciseError } = useSWR(
+        token ? [`${BACKEND_URL}/exercises/getall/${sid}`, token] : null,
+        ExerciseServices.getExerciseDetail
+    );*/
+
     // Create Form Hooks
     const { register: registerSection, handleSubmit: handleSectionUpdate, formState: { errors: sectionErrors } } = useForm<Section>();
-    const { register: registerExercise, handleSubmit: handleExerciseAdd, formState: { errors: exerciseErrors } } = useForm<Exercise>();
-
-    // Submit Handlers for function
-    const onExerciseAdd: SubmitHandler<Exercise> = data => addExercise(data);
-    const onSectionSave: SubmitHandler<Section> = data => saveSection(data);
-
-    // SubmitHandler: Add new exercise to section
-    const addExercise = async (data: Exercise) => {
-        const response = await ExerciseServices.addExercise(data, token, sid)
-        const addedExercise = response.data
-        sectionData.exercises.push(addedExercise)
-        setExercises(sectionData.exercises)
+  
+ /**
+     * Delete section and redirect to course edit page
+     * Uses window.location.href to redirect instead of navigate, as navigate doesn't update the page
+     * 
+     * @param sid The section id
+     * @param token The user token
+     */
+ const deleteSection = async () => {
+    const response = await SectionServices.deleteSection(sid, token);
+    const status = response.status
+
+    if (status >= 200 && status <= 299) {
```

Comment:
This way of filtering out errors, does it work? Or are there any status where you would get something that is not correct?

Reply:
I don't know...